src/sources/soundsourcemp3.cpp

```diff
@@ -600,34 +638,38 @@ SINT SoundSourceMp3::readSampleFrames(
             const SINT madSynthOffset =
                     m_madSynth.pcm.length - m_madSynthCount;
             DEBUG_ASSERT(madSynthOffset < m_madSynth.pcm.length);
-            if (isChannelCountMono()) {
+            const SINT madSynthChannelCount = m_madSynth.pcm.channels;
+            DEBUG_ASSERT(0 < madSynthChannelCount);
+            DEBUG_ASSERT(madSynthChannelCount <= getChannelCount());
+#ifndef QT_NO_DEBUG_OUTPUT
+            if (madSynthChannelCount != getChannelCount()) {
+                qDebug() << "Reading MP3 data with different number of channels"
+                        << madSynthChannelCount << "<>" << getChannelCount();
+            }
+#endif
+            if (1 == math_min(madSynthChannelCount, getChannelCount())) {
```

Comment:
I think this deserves some more comments. I an not sure where, I am still trying to understand. Please correct me: 
"Here we are, if we have decoded a mono sample. This can be either intended or a faulty frame"